crates/controller/src/window.rs

```diff
@@ -0,0 +1,44 @@
+use bevy::{
+    prelude::*,
+    window::{CursorGrabMode, WindowMode},
+};
+use de_conf::Configuration;
+use de_core::state::AppState;
+
+pub struct WindowManagementPlugin;
+
+impl Plugin for WindowManagementPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_systems(
```

Comment:
I think that `add_systems()` returns app, so it is possible to chain these like this:

```rust
app.add_systems(
    OnExit(AppState::AppLoading),
    update_window_mode_based_on_config,
).add_systems(Update, toggle_fullscreen);
```